### ISSUE TITLE
Exposed services are not accessible using external IP

### ISSUE THREAD

**Issue by mfaizanse** [Azure/acs-engine#369]
Hi,
I created a windows +k8s cluster on Azure Container Service. SSH to the master node and created a deployment using:
```
kubectl create deployment iis --image=microsoft/iis
kubectl expose deployment iis --port=80
kubectl edit svc/iis
```
Changed the type from ClusterIP to LoadBalancer and saved it. I got the external IP from kubectl get svc and opened the IP on browser and it worked.

Then I deleted the service & deployment and followed the same procedure to deploy and expose my own container. I got the external ip, then I curl _external-ip_ in the master SSH terminal and it returned the response but when i opened this same external IP on my browser, its not reachable.
 
I tried the above mentioned commands to test iis container again and same issue for this as well now.

There is a same issue on kubernetes git - [link](https://github.com/kubernetes/kubernetes/issues/42151)

**Comment by colemickens** [comment]
cc: @anhowe @JiangtianLi See upstream issue, sounds like there are enough reports that we should look into this. Can one of you take a look?

**Comment by mfaizanse** [comment]
I deployed a new cluster and exposed a service and its accessible. I am not sure what was the issue but there is one thing I have noted that when I was running curl Node IP:NodePort, I was getting connection refused but now in newly deployed cluster i am getting response.

**Comment by JiangtianLi** [comment]
@mfaizanse If you RDP to Windows node, what is State when you run
netsh advf show publicprofile
on the old and new deployment?

**Comment by mfaizanse** [comment]
How to RDP to the windows node? I tried using the public ip but was unable
to rdp.

**Comment by JiangtianLi** [comment]
You can follow the instructions at https://github.com/Azure/acs-engine/blob/master/docs/kubernetes.windows.md

**Comment by mfaizanse** [comment]
okay. I have deleted the old deployment. But you can replicate the problem at your side using the same sequence:
Create & deploy cluster using examples/windows/kubernetes.json
1. kubectl create deployment iis --image=microsoft/iis
2. kubectl expose deployment iis --port=80
3. kubectl edit svc/iis
4 . Change to LoadBalancer
5 . Delete service and deployment

Now, repeat the first 4 steps and this time you won't be able to access exposed service.

P.S. I have tried the same sequence today again just to make sure but I have also deleted that cluster too.

**Comment by JiangtianLi** [comment]
@mfaizanse Thanks for the detailed repro steps. I have a cluster that can repro. I will investigate and update.

**Comment by mfaizanse** [comment]
@JiangtianLi How to allow a container running in a pod to communicate with the outside world?
I am trying to ping www.google.com  from inside a container but getting request timeout.

**Comment by wangtt03** [comment]
@mfaizanse the ping action is blocked. you can `docker exec -it xxx cmd` to attach the container and run `powershell && wget http://www.google.com -OutFile xxx.txt` and it works.

**Comment by mfaizanse** [comment]
@wangtt03 Got it. Thanks

**Comment by yangl900** [comment]
@mfaizanse not sure if anyone find the root cause, but a similar problem is found on Linux cluster too. See issue #446 .  Worth checking the load balancer of the exposed service, if it's in 'Failed' state, then likely same issue.

**Comment by mfaizanse** [comment]
@JiangtianLi What is the status of this issue?

**Comment by JiangtianLi** [comment]
@mfaizanse This has been resolved per fix in v1.5.3 kube-proxy/kubelet Windows binaries.

**Comment by mfaizanse** [comment]
Thank you @JiangtianLi . I have tried multiple deployments and services and its working now.
